rules/aip0121/resource_must_support_list_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0121
+
+import (
+	"testing"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/testutils"
+)
+
+// TestResourceMustSupportList tests the resourceMustSupportList
+// lint rule by declaring a service proto, then declaring a
+// google.api.resource message, then declaring non-List
+// methods.
+func TestResourceMustSupportList(t *testing.T) {
+	for _, test := range []struct {
+		name     string
+		RPCs     string
+		problems testutils.Problems
+	}{
+		{"ValidCreateList", `
+			rpc ListBooks(ListBooksRequest) returns (ListBooksResponse) {};
+			rpc CreateBook(CreateBookRequest) returns (Book) {};
+		`, nil},
+		{"ValidCreateListLRO", `
+			rpc ListBooks(ListBooksRequest) returns (ListBooksResponse) {};
+			rpc CreateBook(CreateBookRequest) returns (google.longrunning.Operation) {
+				option (google.longrunning.operation_info) = {
+					response_type: "Book"
+				};
+			};
+		`, nil},
+		{"ValidListGet", `
+			rpc GetBook(GetBookRequest) returns (Book) {};
+			rpc ListBooks(ListBooksRequest) returns (ListBooksResponse) {};
+		`, nil},
+		{"ValidUpdateList", `
+			rpc ListBooks(ListBooksRequest) returns (ListBooksResponse) {};
+			rpc UpdateBook(UpdateBookRequest) returns (Book) {};
+		`, nil},
+		{"InvalidCreateOnly", `
+			rpc CreateBook(CreateBookRequest) returns (Book) {};
+		`, []lint.Problem{
+			{Message: `resource "library.googleapis.com/Book"`},
+		}},
+		{"InvalidCreateOnlyLRO", `
+			rpc CreateBook(CreateBookRequest) returns (google.longrunning.Operation) {
+				option (google.longrunning.operation_info) = {
+					response_type: "Book"
+				};
+			};
+		`, []lint.Problem{
+			{Message: `resource "library.googleapis.com/Book"`},
+		}},
+		{"InvalidUpdateOnly", `
+			rpc UpdateBook(UpdateBookRequest) returns (Book) {};
+		`, []lint.Problem{
+			{Message: `resource "library.googleapis.com/Book"`},
+		}},
+		{"InvalidGetOnly", `
+			rpc GetBook(GetBookRequest) returns (Book) {};
+		`, []lint.Problem{
+			{Message: `resource "library.googleapis.com/Book"`},
+		}},
+		{"IgnoreSingleton", `
```

Comment:
```suggestion
		{"ValidIgnoreSingleton", `
```

nit really, but most other tests only categorize under `Valid` and `Invalid`, not `Ignore`.